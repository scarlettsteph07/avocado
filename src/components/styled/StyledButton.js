import styled from '../../styled'

export const StyledButton = styled.button`
  background-color: ${({ theme }) => theme.textPrimary};
  background-image: none;
  background-position: 20px center;
  background-repeat: no-repeat;
  background-size: 15px 14px;
  box-shadow: none;
  box-sizing: border-box;
  color: black;
  display: inline-block;
  font-family: ${({ theme }) => theme.baseFont};
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.21;
  letter-spacing: 2.8px;
  text-align: center;
  color: ${({ theme }) => theme.white};
  letter-spacing: 1.5px;
  padding: 10px 15px 10px 15px;
  text-align: center;
  text-transform: uppercase;
  text-decoration: none;
  margin: 0 auto 0 0;
  height: 35px;
  &:hover {
    color: ${({ theme }) => theme.textPrimary};
    box-shadow: inset 0px 0px 0px 2px
      ${({ theme }) => theme.textPrimary};
    background-color: ${({ theme }) => theme.transparent};
  }
`
