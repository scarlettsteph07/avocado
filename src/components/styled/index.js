export * from './StyledButton'
export * from './StyledSlatInner'
export * from './StyledSlatOuter'
