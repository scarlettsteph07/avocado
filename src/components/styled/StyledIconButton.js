import styled from '../../styled'

export const StyledIconButton = styled.button`
  width: 20px;
  height: 20px;
  margin: 31.25px 0;
  background-image: url(/svg/icon--plus.svg);
  background-repeat: no-repeat;
  background-size: 20px;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  @media (min-width: ${({ theme }) => theme.devices.tablet}) {
    width: 40px;
    height: 40px;
    background-size: 40px;
  }
`
