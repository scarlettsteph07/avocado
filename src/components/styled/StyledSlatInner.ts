import styled from '../../styled'

export const StyledSlatInner = styled.div`
  margin: 0 20px;
  overflow: hidden;
  padding: 25px 0;
  @media (min-width: ${({theme}) => theme.large.start}) {
    margin: auto;
    width: 100%;
    max-width: ${({theme}) => theme.large.inner};
  };
  @media (min-width: ${({theme}) => theme.xlarge.start}) {
    width: 100%;
    max-width: ${({theme}) => theme.xlarge.inner};
  };
`;
