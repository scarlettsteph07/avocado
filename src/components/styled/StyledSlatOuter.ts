import styled from '../../styled'

export const StyledSlatOuter = styled.section`
  margin: auto;
  overflow: hidden;
  font-family: ${({ theme }) => theme.baseFont};
  font-weight: normal;
  width: 100%;
  @media (min-width: ${({ theme }) => theme.xlarge.inner}) {
    max-width: ${({ theme }) => theme.xlarge.outer};
  }
`
