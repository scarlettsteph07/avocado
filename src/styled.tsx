import styled, { CreateStyled } from '@emotion/styled'

type Theme = {
  small: {
    outer: string,
    inner: string,
    start: string,
    end: string,
  },
  medium: {
    inner: string,
    start: string,
    end: string,
  },
  large: {
    before: string,
    inner: string,
    start: string,
    end: string,
  },
  xlarge: {
    outer: string,
    inner: string,
    start: string,
  },
  devices: {
    desktop: string,
    tablet: string,
  },

  baseFont: string,
  secondaryFont: string,

  textPrimary: string,
  transparent: string,
  transparentOverlay: string,
  semiTransparentOverlay: string,
  black: string,
  white: string,
  emerald: string,
  lightGray: string,
}

export default styled as CreateStyled<Theme>