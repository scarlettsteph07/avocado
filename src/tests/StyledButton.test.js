import React from 'react'
import serializer from 'jest-emotion'
import renderer from 'react-test-renderer'

import { StyledButton } from '../components/styled/StyledButton'

expect.addSnapshotSerializer(serializer)

describe('StyledButton', () => {
  test('Button renders correctly', () => {
    expect(
      renderer.create(<StyledButton>Reset!</StyledButton>).toJSON(),
    ).toMatchSnapshot()
  })
})
