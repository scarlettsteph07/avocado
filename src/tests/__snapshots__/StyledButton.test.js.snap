// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.emotion-0 {
  background-image: none;
  background-position: 20px center;
  background-repeat: no-repeat;
  background-size: 15px 14px;
  box-shadow: none;
  box-sizing: border-box;
  color: black;
  display: inline-block;
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.21;
  -webkit-letter-spacing: 2.8px;
  -moz-letter-spacing: 2.8px;
  -ms-letter-spacing: 2.8px;
  letter-spacing: 2.8px;
  text-align: center;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  padding: 10px 15px 10px 15px;
  text-align: center;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0 auto 0 0;
  height: 35px;
}

.emotion-0:hover {
  box-shadow: inset 0px 0px 0px 2px;
}

<button
  className="emotion-0"
>
  Reset!
</button>
`;

exports[`StyledButton Button renders correctly 1`] = `
.emotion-0 {
  background-image: none;
  background-position: 20px center;
  background-repeat: no-repeat;
  background-size: 15px 14px;
  box-shadow: none;
  box-sizing: border-box;
  color: black;
  display: inline-block;
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.21;
  -webkit-letter-spacing: 2.8px;
  -moz-letter-spacing: 2.8px;
  -ms-letter-spacing: 2.8px;
  letter-spacing: 2.8px;
  text-align: center;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  padding: 10px 15px 10px 15px;
  text-align: center;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0 auto 0 0;
  height: 35px;
}

.emotion-0:hover {
  box-shadow: inset 0px 0px 0px 2px;
}

<button
  className="emotion-0"
>
  Reset!
</button>
`;
